csp_billing_adapter/adapter.py

```diff
@@ -243,11 +242,12 @@ def main() -> None:
         time.sleep(config.query_interval)  # wait 1 cycle for usage data
 
         while True:
-            now = event_loop_handler(pm.hook, config, log)
-            log.info("Processed event loop at %s", date_to_string(now))
+            start = event_loop_handler(pm.hook, config, log)
```

Comment:
This is coming  from the middle of the event loop handler function. If any metering happens it could get skewed. Could the sleep time be calculated based on the next wake up time? ((now + interval) - now).total_seconds()

Reply:
Yup, my bad.

Reply:
Actually no this is correct, as I moved the get_now() in the event_loop_handler() to the start of that function, similar to what is being done in PR #79, so the return time from the event_loop_handler() should be the current start time of the event loop run.

I can make this more explicit here, and do an additional get_now() in the main loop... But in that case maybe it would be better to change the event_loop_handler() to take time start time as a parameter instead of immediately getting it again?